x-pack/plugins/cloud_security_posture/public/pages/findings/findings_container.tsx

```diff
@@ -4,31 +4,58 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import React from 'react';
+import React, { useMemo } from 'react';
 import { EuiSpacer, EuiTitle, useEuiTheme } from '@elastic/eui';
 import { css } from '@emotion/react';
 import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
 import { FindingsTable } from './findings_table';
 import { FindingsSearchBar } from './findings_search_bar';
 import * as TEST_SUBJECTS from './test_subjects';
 import type { DataView } from '../../../../../../src/plugins/data/common';
 import { SortDirection } from '../../../../../../src/plugins/data/common';
 import { useUrlQuery } from '../../common/hooks/use_url_query';
 import { useFindings, type CspFindingsRequest } from './use_findings';
+import { FindingsGroupBySelector } from './findings_group_by_selector';
+import { INTERNAL_FEATURE_FLAGS } from '../../../common/constants';
+
+export type GroupBy = 'none' | 'resourceType';
 
 // TODO: define this as a schema with default values
-const getDefaultQuery = (): CspFindingsRequest => ({
+const getDefaultQuery = (): CspFindingsRequest & { groupBy: GroupBy } => ({
```

Comment:
maybe we should add `groupBy` as an optional parameter inside `CspFindingsRequest`, or even as a required param with default `none` as value.

Reply:
yes, avoiding inlining is something i'd like to do, along with some type improvements to make it more readable. 

the `groupBy` isn't actually part of the request, it's a flag for us to know what query to send, as opposed to `CspFindingsRequest` which are arguments we pass to `es` (not flags), so adding it there would be confusing. all of this is to say - preferable to do in upcoming PRs, as more changes are needed. this PR includes the simplest thing to only accomplish the UI part.